### ISSUE TITLE
Deploying to non-AWS platform?

### ISSUE THREAD

**Issue by mbrowne** [enhance-dev/enhance-ssr#64]
Hi, what considerations should I be aware of if I want to deploy an Enhance app to another platform like Cloudflare Workers or Google Cloud? It seems like in theory, the output should just be JS functions that can run anywhere, but I didn't see anything in the docs talking about building/deploying without Architect.

**Comment by brianleroux** [comment]
Hey @mbrowne we are focused on AWS, which is a pretty good default fwiw, but would absolutely welcome a PR that opens up support for deploying to other cloud vendors. We started with AWS because it was the only story for serverless for a long time and remains the market leader for those technologies which we depend on.

I like the theory of going multi cloud but less confident about the practice. As you mentioned, Enhance is built on https://arc.codes which deploys using infra-as-code (IaC) solution called CloudFormation to Lambda running Node. Cloudflare workers aren't Node (tho they often claim compatibility the truth is less simple) and don't have a very good IaC story. GCP does support Node but last I checked they also do not have a very good IaC story. I know both CF and GCP can work with Terraform but with the recent licensing uncertainty I do not think its a very good choice for IaC personally.

Anyhow, open to working with you to find a solution for other clouds. Or, if you interested, I'd also be happy to get you setup on AWS.

**Comment by mbrowne** [comment]
Thanks for the explanation. I was wondering if maybe there was some more platform-agnostic code behind the scenes that I could self-deploy, with the end result being something like Remix that can be deployed on any kind of cloud function or even a Docker container (although serverless is preferable). Having the deployment process officially supported by Enhance tooling is less important. 

At this point we are only in the evaluation stage, and this would be for one of our apps. It's discrete enough from our other apps that in theory we could use AWS for this one even though the rest of our infrastructure is in Google Cloud and Cloudflare. So if we decide to go with Enhance, maybe we'll just do that. But it does seem like it might be a shame for this framework to remain locked in to one vendor...that might be more of an issue for others than it is for my company (and even for us, we'd have to make sure that the ingress/egress costs for making this app's data available to the rest of our stack would make sense for us).

**Comment by macdonst** [comment]
That being said you could always call the `@enhance/ssr` [package](https://github.com/enhance-dev/enhance-ssr) directly from a JS environment of your choice. You will lose out on the file based routing and API paths. Those are things you'd have to code up yourself.

**Comment by mbrowne** [comment]
@brianleroux you make a lot of good points about the general goals of the framework, which I didn't fully realize. It might help if I explain more about the use case...this is for a relatively simple blog-like site with thousands of articles that needs to load extremely fast and handle occasional spikes of high traffic. (The site is currently running on WordPress and the JS it's using should really be rewritten.) So I doubt it will need anything more than GET/POST. I'm still not sure if Enhance is the best fit - it's cool that it has all those additional features though :)

@macdonst I might try that out, thanks. Since we're already using Remix for our other apps, I wonder if there might be an opportunity to use remix [resource routes](https://remix.run/docs/en/main/guides/resource-routes) in combination with `@enhance/ssr`. If anything fruitful comes out of that idea, I'll post about it here.

**Comment by macdonst** [comment]
@mbrowne interesting use case. I'd love to jump on a call with you to discuss it in more detail. We run our [blog](https://begin.com/blog) on Enhance (obviously) so we know it works well for blog type sites. I've been playing around with WP more lately so it would be cool if we could find a path for this to work for you.

**Comment by mbrowne** [comment]
@macdonst Thanks, I might not have time for a call until next week, but I joined the discord (username `mbrowne`).

**Comment by mbrowne** [comment]
It looks like rendering enhance components using remix would be doable... I'm sure that the client-side part of the web components implementation is where things would get more complicated (setting up the compilation process for that), but for server-side rendering it was pretty straightforward.

Here's a POC:
https://github.com/mbrowne/remix-enhance-experiment

See http://localhost:3000/blog

**Comment by brianleroux** [comment]
fun stuff. @mbrowne what does this buy you that the regular path with Enhance doesn't have ?

**Comment by mbrowne** [comment]
@brianleroux Just the ability to deploy it anywhere and not only with AWS. The rest of our apps are all running in kubernetes in Google Cloud, and we're probably going to move out main frontend to Cloudflare workers. So AWS doesn't fit into our stack as easily, and would come with more network overhead.

**Comment by brianleroux** [comment]
ok, cool, so…if your app only does GET/POST I think its a pretty small lift to get regular Enhance working on Docker (and probably CF Workers tho we have less desire to support them). bonus: no build step. 

is that something you'd use ?

**Comment by mbrowne** [comment]
> is that something you'd use ?

In theory, yes. We are wrapping up our project just doing everything in Wordpress right now, so for the moment my POC is an experiment to hopefully come back to in the future for a performance boost. But I imagine that others would find this useful as well, for similar blog-type sites that don't need more than GET/POST.

**Comment by brianleroux** [comment]
cool, here's an example dockerizing enhance https://github.com/brianleroux/enhance-example-docker

will chat w the team about adding to the docs

**Comment by mbrowne** [comment]
So simple! Thanks. I didn't realize it would be ok to use the `enhance dev` command to run an app in production.

**Comment by brianleroux** [comment]
Yeah it's just a bare bones Node server so it's fine as long as you stick to http stuff

**Comment by mbrowne** [comment]
Do the docs still need to be updated, or was that already done? Happy to close this issue if no further action is needed right now.

**Comment by evankanderson** [comment]
I'm using this pattern as well, but I only found it by searching https://github.com/enhance-dev for the keyword "Kubernetes".  While Lambda is a fine tool for some use cases, the ability to build into a container as an added deployment mechanism opens up a bunch of other possibilities (for example, my staging setup uses a Raspberry Pi exposed via Tailscale -- adding AWS to the mix would be a big lift, while another docker container is not).
